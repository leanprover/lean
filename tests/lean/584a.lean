variables (A : Type) [H : inhabited A] (x : A)
include H

definition foo := x
check foo  -- A and x are explicit

variables {A x}
definition foo' := x
check @foo' -- A is explicit, x is implicit

open nat

check foo nat 10

definition test : foo' = (10:nat) := rfl

set_option pp.implicit true
print test
