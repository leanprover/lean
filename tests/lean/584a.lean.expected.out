foo : Π (A : Type) [H : inhabited A], A → A
foo' : Π {A : Type} [H : inhabited A] {x : A}, A
foo ℕ 10 : ℕ
definition test : ∀ {A : Type} [H : inhabited A], @foo' ℕ nat.is_inhabited (@has_add.add ℕ nat_has_add 5 5) = 10 :=
λ (A : Type) (H : inhabited A), @rfl ℕ (@foo' ℕ nat.is_inhabited (@has_add.add ℕ nat_has_add 5 5))
